### PR TITLE
Added `thickHugeFungusGrowthFix` rule

### PR DESCRIPTION
**Pull request by Ghoulboy78**
A suggestion by ncolyer in Carpet Discord.
Basically when generating the world, there is as 6% chance for huge fungi to spawn with a 3x3 trunk. This rule currently makes all nether trees spawn 3x3 just so I could quickly debug it.

But should I make it so when this rule is enabled all nether trees spawn 3x3, or should I make it so when it's enabled then 6% of nether trees spawn with a 3x3 trunk?

### CODE REVIEW

**Comment by ncolyer11** [comment]
im still on the side of all player planted have a 100% chance of genning, while, to not disturb world gen, non-planted fungus have a 6% chance of genning

**Comment by Ghoulboy78** [comment]
ah crap forgot about world gen. Rn if carpet rule is active then all generated trees will be fat.

**Comment by Ghoulboy78** [comment]
Well if that bug is fixed then this issue will be obsolete, so ig I can rename it to `thickHugeFungusGrowthFix`

**Comment by Ghoulboy78** [comment]
@altrisi this guy looks ready to go

File: src/main/java/carpet/CarpetSettings.java [hunk 1]
```diff
@@ -1076,4 +1076,11 @@ public String description() {
     )
     public static int sculkSensorRange = 8;
 
+    @Rule(
+            desc = "Allows to grow nether trees with 3x3 base with bonemeal",
+            extra = {"Setting to 'all' will make all nether fungi grow into 3x3 trees", "Setting to 'vanilla' will make 6% of all nether fungi grow into 3x3 trees", "(this being consistent with worldgen)", "Fixes [MC-215169](https://bugs.mojang.com/browse/MC-215169)."},
+            options = {"false", "all", "vanilla"},
+            category = {SURVIVAL, BUGFIX}
+    )
+    public static String thickHugeFungusGrowthFix = "false";
```
Comment (altrisi) [inline src/main/java/carpet/CarpetSettings.java:1085]: This should be an enum for faster checks at runtime, string comparison is slow.

**Review by altrisi — APPROVED** [review]
Looks good other than the comment, I've also been so unlucky with getting the 6% in testing lol (finally got it).

File: src/main/java/carpet/CarpetSettings.java (hunk 1 above)
Comment (altrisi) [reply src/main/java/carpet/CarpetSettings.java:1085]: Something like:
```java
enum FungusFixMode {
    FALSE, VANILLA, ALL;
}
```

File: src/main/java/carpet/CarpetSettings.java [hunk 2]
```diff
@@ -1076,4 +1076,15 @@ public String description() {
...
     public static int sculkSensorRange = 8;
 
+    public enum FungusFixMode {
+        FALSE, VANILLA, ALL;
+    }
+
+    @Rule(
+            desc = "Allows to grow nether trees with 3x3 base with bonemeal",
+            extra = {"Setting to 'all' will make all nether fungi grow into 3x3 trees", "Setting to 'vanilla' will make 6% of all nether fungi grow into 3x3 trees", "(this being consistent with worldgen)", "Fixes [MC-215169](https://bugs.mojang.com/browse/MC-215169)."},
+            options = {"false", "all", "vanilla"},
```
Comment (altrisi) [inline src/main/java/carpet/CarpetSettings.java:1086]:
```suggestion
```

**Pull request merged**